src/diffusers/pipelines/alt_diffusion/pipeline_alt_diffusion.py

```diff
@@ -51,6 +51,23 @@
 """
 
 
+# Copied from diffusers.pipelines.stable_diffusion.pipeline_stable_diffusion.rescale_noise_pred
+def rescale_noise_pred(noise_pred, noise_pred_text, guidance_rescale=0.0):
```

Comment:
I think this is quite a weird design. Let's move classifier free guidance also into this function? What I mean is:

Because currently argument `noise_pred` is confusing. Does it mean unconditional prediction or prediction after cfg (latter is correct but the name kinda imply the prior).

So let's move cfg also into this function.
```python
def classifier_free_guidance(pred_pos, pred_neg, guidance_weight, guidance_rescale=0):
    # Apply classifier-free guidance.
    pred_cfg = pred_neg + guidance_weight * (pred_pos - pred_neg)

    # Apply guidance rescale. # From paper [Common Diffusion Noise Schedules 
    # and Sample Steps are Flawed](https://arxiv.org/pdf/2305.08891.pdf), See section 3.4.
    if guidance_rescale != 0:
        std_pos = pred_pos.std(dim=list(range(1, pred_pos.ndim)), keepdim=True)
        std_cfg = pred_cfg.std(dim=list(range(1, pred_cfg.ndim)), keepdim=True)
        # Fuse equation 15,16 for efficient computation.
        pred_cfg *= guidance_rescale * (std_pos / std_cfg) + (1 - guidance_rescale)
    
    return pred_cfg
```

We wrote equation 15,16 for simpler understanding, but the computation can be fused for more efficient computation.

Reply:
If ok, I'd like to keep them separated for to better seperate classifier free guidance from the rescaling. I don't think we really loose in performance here tbh